### ISSUE TITLE
Configuration of "Exclude pages" with a depth (x+y) leads to an error

### ISSUE THREAD

**Issue by kraemer-igroup** [AOEpeople/crawler#777]
## Bug Report

**Current Behavior**
If the configuration of "Exclude pages" is saved with a depth, like "9+5" (page uid: 9, depth: 5 levels) it immediately leads to the following error:
_Argument 1 passed to AOE\Crawler\Service\ConfigurationService::expandPidList() must be of the type array, null given, called in /.../typo3conf/ext/crawler/Classes/Service/ConfigurationService.php on line 195_

A config like "9,10" is working properly. Despite the error, the record gets saved. But the error comes again when trying to index pages with this setting.

**Expected behavior/output**
No error message when adding a depth.

**Steps to reproduce**
* Create a new crawler configuration or open an existing one
* Write "9+5" in the text field for "Exclude pages"
* Save the record

**Environment**
- Crawler version(s): 10.0.0
- TYPO3 version(s): 10.4.19
- Is your TYPO3 installation set up with Composer (Composer Mode): no

**Possible Solution**
`static $treeCache` seems to be null (https://github.com/AOEpeople/crawler/blob/main/Classes/Service/ConfigurationService.php#L173)

**Additional context**
I am using crawler with `indexed_search`.

**Comment by tomasnorre** [comment]
Thanks for your report @kraemer-igroup 

I'm currently looking through the manual.

> If the configuration of "Exclude pages" is saved with a depth, like "9+5" (page uid: 9, depth: 5 levels) it immediately leads to the following error:

Where do you find the information that this is possible? I'm pretty sure you're right, but don't find the documentation on it right now.

**Comment by kraemer-igroup** [comment]
Thanks for your reply and for looking into this @tomasnorre!
Yes, this is not mentioned within the manual (anymore?), but I often used this and it still is respected in the code which you found.

**Comment by tomasnorre** [comment]
Do you have any information on in which version it worked last time?

**Comment by kraemer-igroup** [comment]
Not exactly, but it still worked with version 9.2.2 . With the versions between this and version 10.0.0 I have no experiences.

**Comment by tomasnorre** [comment]
Thanks that already a big help. I think I know where the problem is. Will get back to you when I have something for testing.

**Comment by kraemer-igroup** [comment]
Looking forward to it! But at first I am on vacation until Sep. 6 ;-)

**Comment by tomasnorre** [comment]
Enjoy your vacation. Pretty sure I'll have a fix till then.

**Comment by tomasnorre** [comment]
Can you please check if this change works for you? 

```diff
-        static $treeCache;
+        static $treeCache = [];
```

In the `expandExcludeString()`-function

**Comment by tomasnorre** [comment]
Hi @kraemer-igroup 

I have been looking at little more into this, and wanted to debug this. 
I cannot reproduce this on the Devbox setup for the Crawler.

Neither with `uid+` nor `uid+depth`

Do you have any idea, what could be the diff between your setup and the Crawler dev box? 
you can start the devbox for testing if you find the time (after your vacation). 

https://github.com/AOEpeople/crawler/blob/main/CONTRIBUTING.md#devbox

**Comment by kraemer-igroup** [comment]
Thank you!!
